Plugins/Traccc/include/Acts/Plugins/Traccc/Detail/AlgebraConversion.hpp

```diff
@@ -0,0 +1,40 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2023 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+// Acts include(s)
+#include "Acts/Definitions/Algebra.hpp"
+
+// System include(s)
+#include <cstddef>
+
+namespace Acts::TracccPlugin::Detail {
+
+/// @brief Creates a new Acts vector from another vector type.
+template <std::size_t N, typename dvector_t>
+inline Acts::ActsVector<N> newVector(const dvector_t& dvec) {
+  Acts::ActsVector<N> res;
+  for (std::size_t i = 0; i < N; i++) {
+    res(i) = static_cast<Acts::ActsScalar>(dvec[i]);
+  }
+  return res;
+}
+/// @brief Creates a new Acts square matrix from another square matrix type.
+template <std::size_t N, typename matrixNxN_t>
+inline Acts::ActsSquareMatrix<N> newSqaureMatrix(const matrixNxN_t& mat) {
```

Comment:
Are the types actually different, i.e. are we doing float conversions here?
If not, I'd probably implement this using an Eigen map type and Eigen's copy operators.